src/v2/Components/Onboarding/Steps/Budget.tsx

```diff
@@ -27,98 +47,75 @@ const OptionsContainer = styled.div`
   `};
 `
 
-interface State {
-  selection: number | null
+const budgetOptions = {
+  "UNDER $500": 500,
+  "UNDER $2,500": 2500,
+  "UNDER $5,000": 5000,
+  "UNDER $10,000": 10000,
+  "UNDER $25,000": 25000,
+  "UNDER $50,000": 50000,
+  "NO BUDGET IN MIND": 1000000000000,
 }
 
-@track({}, { dispatch: data => Events.postEvent(data) })
-export class BudgetComponent extends React.Component<
-  Props & StepProps & SystemContextProps,
-  State
-> {
-  options = {
-    "UNDER $500": 500,
-    "UNDER $2,500": 2500,
-    "UNDER $5,000": 5000,
-    "UNDER $10,000": 10000,
-    "UNDER $25,000": 25000,
-    "UNDER $50,000": 50000,
-    "NO BUDGET IN MIND": 1000000000000,
-  }
+export const BudgetComponent = props => {
+  const tracking = useTracking()
+  const updateProfile = props.updateUserProfile || updateUserProfile
+  const [selectedOption, setSelectedOption] = useState(null)
 
-  state = {
-    selection: null,
+  const onOptionSelected = (index: number) => {
+    const selection = { selection: Object.values(budgetOptions)[index] }
+    setSelectedOption(selection)
   }
 
-  onOptionSelected = (index: number) => {
-    const selection = { selection: Object.values(this.options)[index] }
-    this.setState(selection)
-  }
+  const options = Object.keys(budgetOptions).map((text, index) => (
+    <SelectableToggle
+      key={index}
+      text={text}
+      onSelect={() => onOptionSelected(index)}
+      selected={selectedOption?.selection === budgetOptions[text]}
+    />
+  ))
 
-  submit() {
-    const priceRangeMax = this.state.selection
+  const submit = () => {
+    const priceRangeMax = selectedOption?.selection
+    if (!priceRangeMax) return
 
-    commitMutation<BudgetUpdateMyUserProfileMutation>(
-      this.props.relayEnvironment,
-      {
-        // TODO: Inputs to the mutation might have changed case of the keys!
-        mutation: graphql`
-          mutation BudgetUpdateMyUserProfileMutation(
-            $input: UpdateMyProfileInput!
-          ) {
-            updateMyUserProfile(input: $input) {
-              user {
-                name
-              }
-            }
-          }
-        `,
-        variables: {
-          input: {
-            priceRangeMin: -1,
-            priceRangeMax,
-          },
-        },
-      }
-    )
+    updateProfile(priceRangeMax, props.relayEnvironment)
 
-    const redirectTo = this.props.redirectTo || "/"
+    const redirectTo = props.redirectTo || "/"
     setTimeout(() => window.location.assign(redirectTo), 500)
```

Comment:
This is a bit of a head twister to read -- so a timeout is set, then code after it is executed within the time window, then the function above already executed code fires (redirect) -- can we make this more sequential so that timeout happens at the end?

Reply:
Happy to refactor this but it's what was already there so it's no worse than when I started, haha. Did you have an idea about how to do it sequentially?

Reply:
Its legitimately confusing... We can put the  `setTimeout` function at the end of `submit`, so that the code reads linearly.